### ISSUE TITLE
How does it work?

### ISSUE THREAD

**Issue by cagnulein** [hesbryce/Stamina-Bar#1]
hi! your project seems exactly what I was looking for!

but how does it work?
I see just this, the bar never appears 

let me know!

![image](https://user-images.githubusercontent.com/48216/189886330-304ac94a-7e02-4e07-80f7-d15627fb3864.jpeg)

**Comment by hesbryce** [comment]
Ciao Roberto,

For some reason the app works properly on my own watch, but it appears others cannot load the Stamina Bar.

I am currently in talks with Apple to get this issue resolved. I'll keep you posted, or notify you via GitHub.

I'm curious to know what your use case for my app is!


Thanks in advance,


Bryce
________________________________
From: Roberto Viola ***@***.***>
Sent: Tuesday, September 13, 2022 6:10 AM
To: hesbryce/Stamina-Bar ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [hesbryce/Stamina-Bar] How does it work? (Issue #1)


hi! your project seems exactly what I was looking for!

but how does it work?
I see just this, the bar never appears

let me know!

[image]<https://user-images.githubusercontent.com/48216/189886330-304ac94a-7e02-4e07-80f7-d15627fb3864.jpeg>

**Comment by cagnulein** [comment]
@hesbryce Hi Bryce! Thanks for the fast answer!
My goal would be create a widget for the stamina like the garmin watches :)

Tomorrow, if it can help you, I can try to build the project on my own and see if the bar will be displayed

**Comment by hesbryce** [comment]
Hi Roberto,

Rebuild
I am open to the idea of you recreating Stamina Bar for Apple Watch<https://www.staminabar.app/>. When I simulate the app on my watch it works, so I'm curious to see if the bar will show up on mine when you re-build it.

Widgets/Complications
Is the image attached what you are looking for? In Xcode/Swift these are called Complications. I have recently added this feature, which is uploaded to the App Store now, but I've yet to update my GitHub. At the moment it is only a static image, and once you press on it, the app launches. I hope to have the Stamina Bar Complication be a live representation of the data inside the app.

I hope this emails helps. Looking forward to hearing from you soon.


Best,
Bryce

________________________________

**Comment by cagnulein** [comment]
> Hi Roberto,
> 
> Rebuild
> I am open to the idea of you recreating Stamina Bar for Apple Watch<https://www.staminabar.app/>. 

The website is yours?

> When I simulate the app on my watch it works, so I'm curious to see if the bar will show up on mine when you re-build it.

Sure!

> 
> Widgets/Complications
> Is the image attached what you are looking for? In Xcode/Swift these are called Complications. 

Yes exactly!

> I have recently added this feature, which is uploaded to the App Store now, but I've yet to update my GitHub. At the moment it is only a static image, and once you press on it, the app launches. I hope to have the Stamina Bar Complication be a live representation of the data inside the app.

Hah right! Now I see it!

> 
> I hope this emails helps. Looking forward to hearing from you soon.

Sure, tomorrow I will give it a go!

**Comment by cagnulein** [comment]
hi @hesbryce i can't build the version on github because all the contents of the HeartRateMesurementView is commented. Is it by mistake?

**Comment by hesbryce** [comment]
Hi Roberto,


I have pushed the most recent version (1.4.3) to GitHub.

It's not by mistake HeartRateMesurementView is commented out. In future builds I deleted that file, as you will see.

Hopefully this helps, cheers.


Bryce


________________________________

**Comment by cagnulein** [comment]
yes @hesbryce this version works! could I ask which is the issue that you submit to apple? I mean they are aware about some issue about this?

**Comment by hesbryce** [comment]
Roberto,

Version 1.4.3, which is on GitHub is the same that is on the App Store.

Does the bar show up for you now? Could you send a picture of your watch if so?

I have let Apple know that the image is not showing up, but they haven't been very helpful.
________________________________

**Comment by cagnulein** [comment]
Hah wait @hesbryce 
As long as I run it from Xcode to my watch, the bar is working.
If now I'm running it without Xcode, it doesn't work anymore. Unbelievable!

**Comment by hesbryce** [comment]
Lol, it has been very frustrating, but Apple seems to be getting back with me on my issue. I'll keep you posted.

Thanks for checking out my repo. You have amazing work yourself.

Cheers,
Bryce
________________________________

**Comment by hesbryce** [comment]
Ciao Roberto,

I want to let you know that I fixed the bug. In the end it was a very simple fix, and the newest version is on the App Store.


Cheers,
Bryce
________________________________

**Comment by cagnulein** [comment]
> In the end it was a very simple fix

It's always easy when you fix it :)

> and the newest version is on the App Store

ok thanks, could I ask to commit the changes in order to see the fix? :D

**Comment by hesbryce** [comment]
Would you be open to meeting virtually? I'd love to hear about your use case for Stamina Bar, and talk about collaborating with you on advancing the app.
________________________________

**Comment by cagnulein** [comment]
Hi @hesbryce actually I'm a bit full of work with https://github.com/cagnulein/qdomyos-zwift
My idea about your project is to add the ability to calculate the stamina as garmin watches do. More detail here https://www.garmin.com/en-US/blog/outdoor/introducing-the-garmin-real-time-stamina-feature/

What do you think?

**Comment by hesbryce** [comment]
Hi @hesbryce<https://github.com/hesbryce> actually I'm a bit full of work with https://github.com/cagnulein/qdomyos-zwift
      I wish you good luck with this project, Roberto





My idea about your project is to add the ability to calculate the stamina as garmin watches do. More detail here https://www.garmin.com/en-US/blog/outdoor/introducing-the-garmin-real-time-stamina-feature/

What do you think?


I am familiar with the link. Do you understand everything about it? For the future of Stamina Bar for Apple Watch, I want to add the ability to calculate stamina as garmin watches do. I'm hopeful that if you have value to add to my project, you'd be open to collaborating with me.



Thanks,

Bryce

________________________________

**Comment by cagnulein** [comment]
first of all we need to calculate the FTP. FTP is 80% of the vo2max that we can read from apple health kit. Let's do this modification first.
Then we can use the FTP value to decrease the stamina over time not just related to the HR that actually it has not a great value unfortunately :)

**Comment by cagnulein** [comment]
i just posted something here https://www.reddit.com/r/Garmin/comments/xnnzkm/which_math_algorithm_there_is_behind_stamina/?utm_source=share&utm_medium=ios_app&utm_name=iossmf

**Comment by hesbryce** [comment]
first of all we need to calculate the FTP. FTP is 80% of the vo2max that we can read from apple health kit. Let's do this modification first.
Then we can use the FTP value to decrease the stamina over time not just related to the HR that actually it has not a great value unfortunately :)

      Thanks for your input- even if it's a little harsh/constructive :). I'm trying to figure out how to get the value of vo2 Max using Swift (so little documentation from Apple or Stackoverflow). It's not as straightforward as Heart Rate, but I'm sure I will figure it out. Any ideas welcome.



i just posted something here https://www.reddit.com/r/Garmin/comments/xnnzkm/which_math_algorithm_there_is_behind_stamina/?utm_source=share&utm_medium=ios_app&utm_name=iossmf

      Thanks for posting this. I will follow closely as well. From my time in university, I reached out to my old professors who are Physiology experts. I hope they can help me understand the calculations as well.


________________________________

**Comment by cagnulein** [comment]
sorry i won't to be rude, i mean, i was thinking that your current version was just a proof of concept for something bigger:)

I know well how to read the vo2max. 

but we need first to understand the mathematical model behind this

**Comment by hesbryce** [comment]
sorry i won't to be rude, i mean, i was thinking that your current version was just a proof of concept for something bigger:)

      No worries. Thank you for being able to see the bigger picture! I believe we are fostering a genuine network, and I hope to continue that.

I know well how to read the vo2max,

   It's great you know how to do this in Swift. I am sure it's pretty simple, right? I think it is a learning curve for me right now to easily read/display HealthKit I want to manipulate.



but we need first to understand the mathematical model behind this


    In the meantime, I will continue to use my network to unravel the mathematical model we're looking for. Good news is that my app has hit 100 downloads! It's a pretty cool feeling to see that.

________________________________

**Comment by hesbryce** [comment]
Also, Garmin's "Stamina Bar" is powered by Firstbeat Analytics. I wanted you to be aware in case you didn't know. I've done some research on them in the past

https://www.linkedin.com/company/firstbeat-technologies-oy/
[https://media-exp1.licdn.com/dms/image/C4E0BAQHVBsM60QBG6w/company-logo_200_200/0/1519897834295?e=2147483647&v=beta&t=4q00AMvBSBulcLsAhdzQ8Qqbt8alUlvYui7x60OuOiI]<https://www.linkedin.com/company/firstbeat-technologies-oy/>
Firstbeat Technologies | LinkedIn<https://www.linkedin.com/company/firstbeat-technologies-oy/>
Firstbeat Technologies | 15,550 followers on LinkedIn. Advanced performance analytics for stress, recovery and exercise. | Firstbeat is the leading provider of physiological analytics and services ...
www.linkedin.com


________________________________

**Comment by cagnulein** [comment]
> Also, Garmin's "Stamina Bar" is powered by Firstbeat Analytics. I wanted you to be aware in case you didn't know.

No I wasn't aware of this! Thanks for the info.

**Comment by cagnulein** [comment]
https://the5krunner.com/2022/01/18/garmin-fenix-7-specifications-opinion-review/

https://github.com/GoldenCheetah/GoldenCheetah/blob/master/src/Metrics/WPrime.cpp
